sycl/include/sycl/ext/intel/experimental/esimd/math.hpp

```diff
@@ -1730,6 +1730,27 @@ ESIMD_INLINE uint64_t rdtsc() {
   return retv.template bit_cast_view<uint64_t>()[0];
 }
 
+/// Performs a fused multiply add computation with three vector operands.
+/// @tparam T type of the vector operands
+/// @tparam N size of the vector operands
+/// @param a First vector function argument.
+/// @param b Second vector function argument.
+/// @param c Third vector function argument.
+/// @return the computation result
+template <typename T, int N>
+ESIMD_INLINE __ESIMD_NS::simd<T, N> fma(__ESIMD_NS::simd<T, N> a,
+                                        __ESIMD_NS::simd<T, N> b,
+                                        __ESIMD_NS::simd<T, N> c) {
+  static_assert(__ESIMD_DNS::is_generic_floating_point_v<T>,
+                "fma only supports floating point types");
+  using CppT = __ESIMD_DNS::element_type_traits<T>::EnclosingCppT;
+  auto Ret = __esimd_fmadd<__ESIMD_DNS::__raw_t<CppT>, N>(
+      __ESIMD_DNS::convert_vector<CppT, T, N>(a.data()),
+      __ESIMD_DNS::convert_vector<CppT, T, N>(b.data()),
```

Comment:
The intrinsic requires floating types, but tfloat32 and bf16 and both int types in IR, but I'm not sure if this is the best way to deal with it. I checked the asm for those types and theres only a single extra fcvt instruction per-vector. If we want to get rid of that we might need a VC intrinsic instead of using the LLVM one, or maybe there's a better way I missed.